File: src/main/java/neo/idlib/containers/CInt.java
package neo.idlib.containers;

public class CInt {
    private int val = 0;

    public CInt(int val) {
        this.val = val;
    }

    public CInt() {
    }

    public int getVal() {
        return val;
    }

    public void setVal(int val) {
        this.val = val;
    }

    public int increment() {
        return this.val++;
    }

    public int decrement() {
        return this.val--;
    }

    public void rightShift(int power) {
        this.val >>= power;
    }

    public void leftShit(int power) {
        this.val <<= power;
    }
}

File: src/main/kotlin/neo/idlib/containers/CInt.kt
package neo.idlib.containers

class CInt {
    private var `val` = 0

    constructor(`val`: Int) {
        this.`val` = `val`
    }

    constructor()

    fun getVal(): Int {
        return `val`
    }

    fun setVal(`val`: Int) {
        this.`val` = `val`
    }

    fun increment(): Int {
        return `val`++
    }

    fun decrement(): Int {
        return `val`--
    }

    fun rightShift(power: Int) {
        `val` = `val` shr power
    }

    fun leftShit(power: Int) {
        `val` = `val` shl power
    }
}